captum/attr/_core/neuron/neuron_integrated_gradients.py

```diff
@@ -30,17 +40,45 @@ def __init__(self, forward_func, layer, device_ids=None):
         NeuronAttribution.__init__(self, forward_func, layer, device_ids)
         GradientAttribution.__init__(self, forward_func)
 
+    @typing.overload
```

Comment:
nit: in order to not repeat typing it would be good to import `typing` everywhere and just call `@overload`  ?

Reply:
Yes, I initially had that, the problem is that flake8 doesn't recognize the overloading appropriately and errors with just @overload, more details are here https://github.com/PyCQA/pyflakes/issues/320. The fix has been added to pyflakes, but isn't yet in the latest version of flake8, once it gets added we can switch back to just @overload.